### PR TITLE
Add basic integration testing infrastructure for validating Problem event emissions

### PR DESCRIPTION
**Pull request by hegyibalint**

### CODE REVIEW

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractTaskRelocationIntegrationTest.groovy [hunk 1]
```diff
@@ -18,11 +18,6 @@ package org.gradle.integtests.fixtures
 
 abstract class AbstractTaskRelocationIntegrationTest extends AbstractIntegrationSpec {
 
-    @Override
-    def setupBuildOperationFixture() {
-        //disable because of a test that is incompatible with the build operation fixture
-    }
-
```
Comment (reinsch82) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractTaskRelocationIntegrationTest.groovy:25]: :question: 
when you don't disable it here,
Shouldn't there be a call to `disableProblemsApiCheck`somewhere?

**Review by reinsch82 — CHANGES_REQUESTED** [review]
a Question

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractTaskRelocationIntegrationTest.groovy (hunk 1 above)
Comment (hegyibalint) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractTaskRelocationIntegrationTest.groovy:25]: It's not enabled by default (for now).
I think we can defer doing this job after we will turn on the checking by default, which is a bit far into the future.

**Review by reinsch82 — APPROVED** [review]
thx

**Pull request merged**